src/Resources/contao/classes/DataContainer.php

```diff
@@ -901,25 +908,62 @@ protected function initPicker()
 			return false;
 		}
 
-		list($this->strPickerTable, $this->strPickerField, $this->intPickerId) = explode('.', \Input::get('target'), 3);
+		$strTarget = \Input::get('target');
+
+		if (\Validator::isInsecurePath($strTarget))
```

Comment:
I see it as a kind of early return, because why continue if we already know that there is a problem?